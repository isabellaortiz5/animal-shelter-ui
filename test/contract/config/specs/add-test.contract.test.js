import { provider } from '../init-pact';
import { Matchers } from '@pact-foundation/pact';
import { AnimalController } from '../../../../controllers';

describe('Animal Service', () => {
    describe('When a request to add an animal is made', () => {
        
        let pact_body = {"name": "Joy",
                        "breed": "Bengali",
                        "gender": "Male",
                        "vaccinated": true}
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                uponReceiving: 'a request to add an animal',
                state: "has animals",
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    headers: {
                        'Content-Type': 'application/json'
                    },
                    body: pact_body
                },
                willRespondWith: {
                    status: 200,
                    headers: {
                        'Content-Type': 'application/json'
                    },
                    body: pact_body
                }
            });
        });
    
        it('Animal registered', async () => {
            const response = await AnimalController.register(pact_body);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(() => provider.finalize());

    })
});

