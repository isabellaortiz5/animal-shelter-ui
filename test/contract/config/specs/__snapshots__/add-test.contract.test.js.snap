// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to add an animal is made Animal registered 1`] = `
Object {
  "breed": "Bengali",
  "gender": "Male",
  "name": "Joy",
  "vaccinated": true,
}
`;
